### PR TITLE
Build and deploy Elodin CLI and simulation onto Aleph

### PR DESCRIPTION
**Pull request by x46085**
By popular request, one can now deploy Elodin directly to the Aleph, and run simulations alongside the flight software for SITL and Sim-HITL use cases.

---

> [!NOTE]
> **High Risk**
> Opening the full TCP/UDP port range when elodin-db firewall is enabled is a major exposure on deployed hardware; otherwise changes are mostly packaging and observability.
> 
> **Overview**
> Adds **on-device Elodin simulation** to Aleph via a new `services.elodin` NixOS module: installs the CLI (Jetson Vulkan/graphics wrapper, JAX built without CUDA), seeds a shared writable **`ELODIN_ASSETS_DIR`** at `/var/lib/elodin/assets`, optionally symlinks packaged examples, and runs a one-shot asset seed service. New Aleph packages wire **`elodin-cli`**, **`elodin-assets`**, and **`elodin-examples`** (video-stream scripts updated to use the system **elodinsink** plugin). The Aleph flake/template import the **`elodin`** module alongside existing flight-software modules.
> 
> When **`services.elodin-db.openFirewall`** is enabled, firewall rules change from opening specific DB ports to allowing **all TCP and UDP ports (1–65535)** for external access—much broader than the previous 2240/2248 behavior.
> 
> Supporting tweaks: **`makeWrapperArgs`** accepts optional **`graphicsWrapperArgs`** for platform-specific CLI wrapping; asset HTTP and editor web-asset fetch paths gain clearer logging/errors; the video-stream example no longer passes **`start_timestamp=0`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b29f2a5c05e410770360b78f6b5f2c8fbc0b851e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: aleph/pkgs/elodin-cli.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  lib,
+  rustToolchain,
+  ...
+}: let
+  # The Aleph sets nixpkgs.config.cudaSupport = true (aleph-cuda.nix), which
+  # flips jax's default to pull the CUDA plugin (-> nccl, unsupported on
+  # aarch64). Build jax without CUDA; the sim uses the cranelift backend and the
+  # renderer uses Vulkan, so GPU jax is not needed on-device.
+  pythonPackages = pkgs.python313Packages.overrideScope (_final: prev: {
+    jax = prev.jax.override {cudaSupport = false;};
```
[the remark this answers is not in the document]
Comment (x46085) [reply aleph/pkgs/elodin-cli.nix:12]: This is a nonsense suggestion, closing.

File: aleph/modules/elodin-db.nix [hunk 2]
```diff
@@ -119,6 +119,6 @@ in {
     };
 
     environment.systemPackages = [elodin-db];
-    networking.firewall.allowedTCPPorts = lib.optionals cfg.openFirewall [2240 2248];
+    networking.firewall.allowedTCPPorts = lib.optionals cfg.openFirewall [2240 2241 2248 9000];
```
[the remark this answers is not in the document]
Comment (x46085) [reply aleph/modules/elodin-db.nix]: Done

File: aleph/modules/elodin-db.nix [hunk 3]
```diff
@@ -119,6 +119,17 @@ in {
...
+        from = 1;
+        to = 65535;
+      }
+    ];
+    networking.firewall.allowedUDPPortRanges = lib.optionals cfg.openFirewall [
+      {
+        from = 1;
+        to = 65535;
+      }
+    ];
```
[the remark this answers is not in the document]
Comment (x46085) [reply aleph/modules/elodin-db.nix:133]: intentional

**Review by anderspitman — APPROVED** [review]
